base/path.jl

```diff
@@ -110,18 +110,17 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
 
 @windows_only realpath(path::AbstractString) = realpath(utf16(path))
 @windows_only function realpath(path::UTF16String)
-    p = UInt32((sizeof(path)>>2) + 1)
+    buf = Array(UInt16, length(path.data))
     while true
-        buflength = p
-        buf = zeros(UInt16,buflength)
         p = ccall((:GetFullPathNameW, "Kernel32"), stdcall,
             UInt32, (Cwstring, UInt32, Ptr{UInt16}, Ptr{Void}),
-            path, buflength, buf, C_NULL)
+            path, length(buf), buf, C_NULL)
         systemerror(:realpath, p == 0)
-        if (p < buflength)
-            resize!(buf, p+1)
-            return utf8(UTF16String(buf))
+        if (p > length(buf))
```

Comment:
Should be `p+1 > length(buf)`, I think.   According to the [documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364963(v=vs.85).aspx), the return value is:

>If the function succeeds, the return value is the length, in TCHARs, of the string copied to lpBuffer, not including the terminating null character.
> If the lpBuffer buffer is too small to contain the path, the return value is the size, in TCHARs, of the buffer that is required to hold the path and the terminating null character.

In particular, note the inconsistency regarding whether the return value includes the terminating null codeunit.